int8 sigmoid op

harouwu opened the pull request:
Summary: Int8 sigmoid ops, test provided. Only supports first axis now

Differential Revision: D12837824

Maratyszcza commented on caffe2/operators/quantized/int8_sigmoid_op.h:
```diff
@@ -0,0 +1,256 @@
+      x * (1 << left_shift), quantized_multiplier);
+}
+
+int CalculateInputRadius(int input_integer_bits, int input_left_shift) {
```
I think I've seen these functions (`QuantizeMultiplierGreaterThanOne`, `MultiplyByQuantizedMultiplierGreaterThanOne`, `CalculateInputRadius`) in several files. Please move them to `int8_utils.h` to avoid duplication

Maratyszcza commented on caffe2/operators/quantized/int8_sigmoid_op.h:
```diff
@@ -0,0 +1,256 @@
+    const int32_t input_left_shift,
+    const int32_t size) {
+  int c = 0;
+#ifdef USE_NEON
```
This macro is never defined. The NEON guarding macro is `INT8_NEON_SIMD`

harouwu committed:
int8 sigmoid op (#13298)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13298

Int8 sigmoid ops, test provided. Only supports first axis now

Reviewed By: newstzpz

Differential Revision: D12837824

fbshipit-source-id: 084f7f31780a629082ea3e9926f3be8807b71067
